### DISCUSSION TITLE
Bad Gateway 502 error

### DISCUSSION THREAD

**Discussion by raviwarrier** [hassio-addons/addon-ssh#244] [Q&A]
Hi Frenck, I was wondering what I am doing wrong. I am a complete noob to all things linux and I am sure I am doing something wrong.

I have my hass on rpi4 and is configured to be accessed by duckdns. I don't know if that matters, but still.

I installed the add-on and started it, but when I click on 'Open Web UI', I get to a page that gives me a 502 Bad Gateway error. I forwarded my port 22 on my router (because that's what I had to do for all my add-ons - AdGuard, Plex, etc.).

I really need to SSH into hass because all the cool tools require me to copy/paste some commands into command line. For example, I am trying to get my rpi sensors to check for temp, cpu and mem usage and this site (https://github.com/Sennevds/system_sensors#lovelace-ui-example) tells me I need to run pip3 and apt-get commands, which I cannot.

I also tried logging into ssh using Putty, but the darn server doesn't allow me to.

I would really like some help here. Please. 

Thanks in advance. And I love the work you have done so far. Thanks for that as well.

Take care, stay safe.
W.

**Comment by frenck** [comment]
What is in the add-on logs?

**Reply by raviwarrier** [reply]
I'm guessing it's a password problem, but I don't know what the password (or for that matter username) is. For other installations on rpi, I used to ssh into rpi with pi@xx.xx.xx.xx and raspberry and then change my password. But since I cannot do that here, I don't know what the default username/password is.

I am sorry for all the trouble. Here is the log you requested.

W

- Log -

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] permissions: applying... 
[fix-attrs.d] permissions: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: SSH & Web Terminal
 SSH & Web Terminal access to your Home Assistant instance
-----------------------------------------------------------
 Add-on version: 8.1.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 5.13  (armv7 / raspberrypi4)
 Home Assistant Core: 2021.4.4
 Home Assistant Supervisor: 2021.04.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] docker.sh: executing... 
[23:25:17] INFO: Docker support has been enabled.
[cont-init.d] docker.sh: exited 0.
[cont-init.d] mosquitto.sh: executing... 
[cont-init.d] mosquitto.sh: exited 0.
[cont-init.d] mysql.sh: executing... 
[cont-init.d] mysql.sh: exited 0.
[cont-init.d] ssh.sh: executing... 
[23:25:19] FATAL: 
[23:25:19] FATAL: Configuration of this add-on is incomplete.
[23:25:19] FATAL: 
[23:25:19] FATAL: Please be sure to set a least a SSH password
[23:25:19] FATAL: or at least one authorized key!
[23:25:19] FATAL: 
[23:25:19] FATAL: You can configure this using the "ssh.password"
[23:25:19] FATAL: or the "ssh.authorized_keys" option in the
[23:25:19] FATAL: add-on configuration.
[23:25:19] FATAL: 
[cont-init.d] ssh.sh: exited 1.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
-----------------------------------------------------------
                Oops! Something went wrong.

 We are so sorry, but something went terribly wrong when
 starting or running this add-on.
 
 Be sure to check the log above, line by line, for hints.
-----------------------------------------------------------
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.

**Reply by raviwarrier** [reply]
I'm sorry, I don't know why the font changed to bold.

**Reply by raviwarrier** [reply]
Oh it worked. I put in the username and password that I log into home assistant with. Yay!

But I have one question: how do I sudo, apt-get, update, pip3? Because it still says 'command not found'.

Sorry if I am making you face-palm yourself with my stupidity.

**Reply by frenck** [reply]
> But I have one question: how do I sudo, apt-get, update, pip3? Because it still says 'command not found'.

You don't as this is not a debian/ubuntu environment. 

Looking at the link provided: <https://github.com/Sennevds/system_sensors#installation>

That will not work and is only usable on system that use a Home Assistant Core installation type, where everything is installed manually in Python directly (which you don't have).

**Reply by raviwarrier** [reply]
Hmmm. Okay. But thank you anyway for all your help. Take care. And have a wonderful day ahead.
